test/signalrclient-e2e-tests/connection_tests.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include "stdafx.h"
+#include <string>
+#include "cpprest\details\basic_types.h"
+#include "cpprest\json.h" 
+#include "connection.h"
+#include "hub_connection.h"
+#include "url.h"
+
+using namespace signalr;
+using namespace utility;
+using namespace std;
+using namespace pplx;
+
+TEST(connection_tests, send_message)
+{
+    auto conn = make_shared<connection>(url + U("raw-connection"));
+    auto message = make_shared<string_t>();
+    auto test_event = make_shared<pplx::event>();
+
+    conn->set_message_received([message, test_event](const string_t& payload)
+    {
+        *message = payload;
+        test_event->set();
+    });
+
+    conn->start()
+        .then([conn](task<void> start_task)
+        {
+            try
+            {
+                start_task.get();
+                ASSERT_EQ(conn->get_connection_state(), connection_state::connected);
+
+                web::json::value obj;
+                obj[U("type")] = web::json::value::number(0);
+                obj[U("value")] = web::json::value::string(U("test"));
+
+                conn->send(obj.serialize());
+            }
+            catch (...)
+            { }
+        });
+
```

Comment:
The way I would write this test (one of the benefits is avoiding handling exceptions in multiple places):

    try
    {
        conn->start().then([conn]()
        {
                ASSERT_EQ(conn->get_connection_state(), connection_state::connected);

                web::json::value obj;
                obj[U("type")] = web::json::value::number(0);
                obj[U("value")] = web::json::value::string(U("test"));
                 return conn->send(obj.serialize());
            }).get();
        }
    catch(...)
    {
        ASSERT_TRUE(false)
    }

- You are blocking on the event so it does not matter if you block the on start and send sequence
- value based continuations are skipped when a previous task throws and we don't really care which of the tasks will throw because neither should so just one handler is enough. 
- I would also consider just hardcoding the payload instead of building it but it's up to you
- connection here does not have to be a shared_ptr here because you are blocking at the end waiting effectively for all continuations to be complete before exiting the function and as a result there is no risk that any of the task will outlive the connection instance. Therefore it can also be captured by reference.